apps/codesearch/deployment.yaml

```diff
@@ -0,0 +1,26 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: codesearch
+  namespace: codesearch
+  labels:
+    app: codesearch
+spec:
+  replicas: 2
+  strategy:
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 1
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: codesearch
+    spec:
+      terminationGracePeriodSeconds: 30
+      containers:
+        - name: codesearch
+          image: k8scode/hound:v0.1.0
```

Comment:
@jimdaga cs.k8s.io is currently running about with a custom build more recent that what's is used to build this image. Unfortunately The hound image on Docker Hub is now unmaintained. you should take a look at `ghcr.io/hound-search/hound:latest`.

Reply:
I just tested updating the original [Dockerfile](https://github.com/dims/k8s-code.appspot.com/blob/master/codesearch.k8s.io/Dockerfile) with the suggested `ghcr.io/hound-search/hound:latest` and things look good.

_(Example with a very short config.json file)_
```
FROM ghcr.io/hound-search/hound:latest

COPY config-demo.json /data/config.json

VOLUME ["/data"]

EXPOSE 8080

ENTRYPOINT ["/go/bin/houndd", "-conf", "/data/config.json", "-addr", ":8080"]
```

```
docker run -it -p 8080:8080 docker.io/dims/hound:v0.1.0
2021/08/18 01:38:12 Searcher started for kubernetes-sigs/krew
2021/08/18 01:38:15 merge 0 files + mem
2021/08/18 01:38:15 1016427 data bytes, 637631 index bytes
2021/08/18 01:38:15 All indexes built!
2021/08/18 01:38:15 running server at http://localhost:8080
```

My questions for you @ameukam are:
1. Are we fine just running the public container direct? or should we be building and storing our own copies for things like container scanning, auditing, etc?
2. IF we build and store our own copy, the code should live under https://github.com/kubernetes/test-infra - yes?

The other thing I'll have to figure out is how to automate the `fetch-repos.py` script so we can update the `config.json` file. It seems like this may need to happen regularly. My first thought was a kubernetes cronjob that runs the python script, and updates a configmap for the `config.json` - But I would like others input into if thats ideal. 

_(And I totally get this all depends on consensus on #2182 👍  )_

Reply:
1. We currently use a weekly build of hound based of their main branch. The `latest` tag is behind what we are running now. We will probably need our owns versions of it. 

2. Totally open to the idea. I don't have specific preferences. We can also use a periodic prowjob that run this script and open a PR if changes are detected.